### PR TITLE
Support for media files metadata sanitizing

### PR DESCRIPTION
**Pull request by davidmz**
New flag on the "Privacy" Settings page: "Remove geolocation and other sensitive metadata from photos and videos you post".

Requires server feature: FreeFeed/freefeed-server#556

### CODE REVIEW

File: src/components/settings/forms/privacy.jsx [hunk 1]
```diff
@@ -80,6 +81,19 @@ export default (function PrivacyForm() {
...
       </section>
 
+      <section className={settingsStyles.formSection}>
+        <h4 id="files">Your files</h4>
+
+        <div className="form-group">
+          <div className="checkbox">
+            <label>
+              <CheckboxInput field={sanitizeMediaMetadata} />
+              Remove geolocation and other sensitive metadata from photos and videos you post
```
Comment (n1313) [inline src/components/settings/forms/privacy.jsx]: Do we sanitize EXIFs for pictures that I upload to comments? Maybe `from photos and videos you upload` is better?

File: src/components/settings/forms/privacy.jsx [hunk 2]
```diff
@@ -108,6 +122,7 @@ function initialValues(userData) {
   return {
     privacy: privacyFlagsToString(userData),
     acceptDirectsFrom: userData.preferences?.acceptDirectsFrom,
+    sanitizeMediaMetadata: userData.preferences?.sanitizeMediaMetadata || false,
```
Comment (n1313) [inline src/components/settings/forms/privacy.jsx]: It's _true_ by default on the server. I think it should be _true_ by default here as well.

**Comment by n1313** [comment]
Should this branch be started from `release`? :)

File: src/components/settings/forms/privacy.jsx (hunk 1 above)
Comment (davidmz) [reply src/components/settings/forms/privacy.jsx]: Yes, all files are sanitized. So yes, may be 'upload' will be better…

File: src/components/settings/forms/privacy.jsx (hunk 2 above)
Comment (davidmz) [reply src/components/settings/forms/privacy.jsx]: I am not sure. In real life, this flag can be undefined only if the server is outdated and, therefore, is NOT sanitize anything. On the other hand, such server will not receive this preferences update.

Comment (n1313) [reply src/components/settings/forms/privacy.jsx]: Will the client read this flag from the server and set this checkbox to match?

**Comment by davidmz** [comment]
Hmm… Shouldn't it?

File: src/components/settings/forms/privacy.jsx (hunk 2 above)
Comment (davidmz) [reply src/components/settings/forms/privacy.jsx]: Yes, the userData.preferences comes from the server

**Comment by n1313** [comment]
I thought that's what we are supposed to do? /cc @indeyets

File: src/components/settings/forms/privacy.jsx (hunk 2 above)
Comment (n1313) [reply src/components/settings/forms/privacy.jsx]: Alright then. Maybe we should (sometime in the future) refactor this and make the client hide preferences that are undefined on the server.

**Comment by davidmz** [comment]
Well, I started it from release, as we speak early
image

**Comment by n1313** [comment]
Oops, my bad, I can't read. Please ignore me.

File: src/components/settings/forms/privacy.jsx (hunk 2 above)
Comment (davidmz) [reply src/components/settings/forms/privacy.jsx]: I still don't think the client should specifically support an outdated server. These `?.` are not really needed here either, they're just already scattered all over the code...

File: src/components/settings/forms/privacy.jsx (hunk 1 above)
Comment (davidmz) [reply src/components/settings/forms/privacy.jsx]: Пофиксил

File: src/components/settings/forms/privacy.jsx (hunk 2 above)
Comment (davidmz) [reply src/components/settings/forms/privacy.jsx]: Я убрал там лишнее, он теперь просто читает preferences без попыток применить интеллект

**Review by n1313 — DISMISSED** [review]

**Pull request merged**